public/app/features/plugins/extensions/utils.tsx

```diff
@@ -102,13 +104,18 @@ const getModalWrapper = ({
   body: Body,
   width,
   height,
-}: PluginExtensionOpenModalOptions) => {
+  pluginId,
+}: PluginExtensionOpenModalOptions & { pluginId: string }) => {
   const className = css({ width, height });
 
   const ModalWrapper = ({ onDismiss }: ModalWrapperProps) => {
+    const { value: isSandboxEnabled } = useAsync(() => shouldLoadPluginInFrontendSandbox({ pluginId }), [pluginId]);
+
     return (
       <Modal title={title} className={className} isOpen onDismiss={onDismiss} onClickBackdrop={onDismiss}>
-        <Body onDismiss={onDismiss} />
+        <div {...(isSandboxEnabled && { 'data-plugin-sandbox': pluginId })} data-testid="plugin-sandbox-wrapper">
```

Comment:
@academo I was wondering if we should be waiting for the `shouldLoadPluginInFrontendSandbox()` to load before rendering anything.

(It could be risky if the sandbox availability check would hang, because then the UI will not render. I think if the sandbox is enabled and we render something before the `shouldLoadPluginInFrontendSandbox()` check would complete, then the worst that can happen is that an extension doesn't have access to elements inside that div for a short timewindow.)